scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
So I think these make sense and should work:

```
correlate(ones(4), ones(2), 'valid')
correlate(ones(2), ones(4), 'valid')
correlate(ones(4,5), ones(2,3), 'valid')
correlate(ones(2,3), ones(4,5), 'valid')
...
```

But these should still raise an exception:

```
correlate(ones(3,2), ones(2,3), 'valid')
correlate(ones(2,3), ones(3,2), 'valid')
```

because `valid` mode requires that one of the arrays "fit inside" the other, or in other words, that only one array is "overhanging" the other.  In this case, both arrays are overhanging each other in different dimensions.